tests/offline.spec.ts

```diff
@@ -1,11 +1,22 @@
-import { test, expect } from "@playwright/test";
+import { test, expect, BrowserContext, Request, Route } from "@playwright/test";
 import { exec } from "child_process";
 import { promisify } from "util";
 
 const asyncExec = promisify(exec);
 const goOnline = async () => await asyncExec("bin/online");
 const goOffline = async () => await asyncExec("bin/offline");
 
+const swWaitForRequest = (context: BrowserContext, url: string) => {
+  return new Promise<Request>((resolve) => {
+    context.route(url, (route: Route) => {
+      if (route.request().serviceWorker()) {
+        resolve(route.request());
+      }
+      return route.fulfill();
+    });
+  });
+};
```

Comment:
The native `page.waitForRequest` doesn't listen to requests in the ServiceWorker, so I had to write my own wrapper.

Based on Playwright's documentation for `PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS`: https://playwright.dev/docs/service-workers-experimental#network-events-and-routing